Sources/PubNubChatComponents/Message List/MessageListViewController.swift

```diff
@@ -82,6 +106,58 @@ open class ChatViewController: CollectionViewComponent {
     })
   }
   
+  open override func onLongPressGestureRecognized(gesture: UIGestureRecognizer) {
+    
+    if gesture.state == .began {
+      
+      // Finds the cell that was pressed
+      guard let affectedCell = collectionView.visibleCells.first(where: {
+        $0.frame.contains(gesture.location(in: collectionView))
+      }) as? MessageListItemCell else {
+        return
+      }
+      
+      let reactionList = affectedCell.reactionListView.allReactions.map() { $0.reaction }
+      let pickerView = AddMessageReactionComponent.DefaultPickerView(reactionList: reactionList)
+      
+      // Creates and configures a view controller responsible for displaying the emoji picker view
+      let viewController = AddMessageReactionComponent(pickerView: pickerView, reactionList: reactionList)
+      viewController.modalPresentationStyle = .custom
+      viewController.transitioningDelegate = self
+      
+      // Captures selection from the emoji picker view
+      viewController.selectedReactionValue.sink(receiveValue: { [weak self] selectedValue in
+        guard
+          let self = self,
+          let indexPath = self.collectionView.indexPath(for: affectedCell),
+          let viewModel = self.viewModel as? MessageListComponentViewModel<ModelData, ManagedEntities>
+        else {
+          return
+        }
+        
+        let message = viewModel.fetchedEntities.object(at: indexPath) as ManagedEntities.Message
+        let buttonComponent = affectedCell.reactionListView.allReactions.first(where: { $0.reaction == selectedValue} )
+                 
+        if let buttonComponent = buttonComponent {
+          self.messageListViewModel?.messageActionTapped?(
+            viewModel,
+            buttonComponent,
+            message, nil
+          )
+        } else {
+          preconditionFailure("Not supported yet")
```

Comment:
is it possible to occur? will it be implemented before pushing to production?

Reply:
It's not possible to occur now because an end-user doesn't have a possibility to provide a different emoji set other than 6 predefined items like 👍, ❤️, 😂, 😲, 😢, 🔥. But if such possibility existed then we would have to implement it. We'll need it but not necessarily now.